provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/client/VlerClient.java

```diff
@@ -0,0 +1,45 @@
+package gov.va.api.health.providerdirectory.service.client;
+
+import gov.va.api.health.providerdirectory.service.AddressResponse;
+
+/** This is the abstraction for communicating with PPMS through their Restful API. */
+public interface VlerClient {
+
+  AddressResponse endpointByAddress(String address);
+
+  /** A request was malformed, such as missing required search parameters. */
+  public static final class BadRequest extends VlerException {
+
+    public BadRequest(String message, Throwable cause) {
+      super(message + ", " + cause.getMessage(), cause);
+    }
+  }
+
+  /** The resource requested was not found. */
+  public static final class NotFound extends VlerException {
```

Comment:
Above comment still stands... the `NotFound`, `SearchFailed` and `BadRequest` exceptions can be refactored into another class and shared with `RestPpmsClient`. (The PPMS exceptions inherit from `PpmsException`, while these inherit from `VlerException`. I'm inclined to say that both parents `PpmsException` and `VlerException` can be deleted, and the `NotFound`, `SearchFailed` and `BadRequest` exceptions can extend from `RuntimeException` directly.